File: sentinelai-backend/app/schemas/log.py
from datetime import datetime
from app.schemas.base import BaseSchema


class LogFileResponse(BaseSchema):
    id: str
    original_filename: str
    source_type: str
    source_name: str | None
    size: int
    mime_type: str | None
    checksum_sha256: str
    status: str
    error_message: str | None
    uploaded_by: str | None
    upload_time: datetime
    processing_started_at: datetime | None
    processing_completed_at: datetime | None
    parsed_entries: int
    failed_entries: int
    tags: list[str] | None


class LogFileListResponse(BaseSchema):
    items: list[LogFileResponse]
    total: int
    page: int
    page_size: int


class LogUploadResponse(BaseSchema):
    id: str
    original_filename: str
    size: int
    checksum_sha256: str
    status: str
    upload_time: datetime
    message: str


class LogStatsResponse(BaseSchema):
    total_files: int
    total_size: int
    total_parsed_entries: int
    by_source_type: dict[str, int]
    by_status: dict[str, int]
    recent_uploads: int
    avg_file_size: float
    storage_used: str

File: sentinelai-frontend/src/types/log.ts
export interface LogFile {
  id: string
  original_filename: string
  source_type: string
  source_name: string | null
  size: number
  mime_type: string | null
  checksum_sha256: string
  status: string
  error_message: string | null
  uploaded_by: string | null
  upload_time: string
  processing_started_at: string | null
  processing_completed_at: string | null
  parsed_entries: number
  failed_entries: number
  tags: string[] | null
}

export interface LogUploadResponse {
  id: string
  original_filename: string
  size: number
  checksum_sha256: string
  status: string
  upload_time: string
  message: string
}

export interface LogFileListResponse {
  items: LogFile[]
  total: number
  page: number
  page_size: number
}

export interface LogStatsResponse {
  total_files: number
  total_size: number
  total_parsed_entries: number
  by_source_type: Record<string, number>
  by_status: Record<string, number>
  recent_uploads: number
  avg_file_size: number
  storage_used: string
}
